### PR TITLE
[BEAM-5931] Update nexmark performance test with dataflow worker jar

### PR DESCRIPTION
**Pull request by HuangLED**
Add worker jars to Nexmark performance test via pipeline options. 

Following local run succeeded (by taking one target out of beam_PostCommit_Java_Nexmark_Dataflow) :

./gradlew :beam-sdks-java-nexmark:run \
  -Pnexmark.runner=":beam-runners-google-cloud-dataflow-java" \
  -Pnexmark.args=" \
     --bigQueryTable=nexmark \
	 --bigQueryDataset=nexmark \
	 --project=$PROJECT \
	 --resourceNameMode=QUERY_RUNNER_AND_MODE \
	 --exportSummaryToBigQuery=true \
	 --tempLocation=$GCS_LOCATION \
	 --runner=DataflowRunner \
	 --numWorkers=4 \
	 --maxNumWorkers=4 \
	 --autoscalingAlgorithm=NONE \
	 --nexmarkParallel=16 \
	 --streaming=true \
	 --suite=STRESS \
	 --manageResources=false \
	 --monitorJobs=true \
	 --enforceEncodability=true \
	 --enforceImmutability=true \
  "

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [X ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by HuangLED** [comment]
R: @lukecwik 
CC: @boyuanzz

**Comment by HuangLED** [comment]
Note that in a update-to-head repo with this PR, I applied a revert-revert, it passes as well with the command line above.

However, as a comparison, in another repo WITHOUT this PR, I applied a revert-revert, the above command also passed.  Could just because there are other test cases in the jenkins spec file. I will try a few more test case if we can reproduce.  

At this point it probably does not matter though, as long as this fix is something should be done and passes.  Just FYI.

File: sdks/java/testing/nexmark/build.gradle [hunk 1]
```diff
@@ -98,7 +103,16 @@ if (shouldProvideSpark) {
 //   -Pnexmark.args
 //       Specify the command line for invoking org.apache.beam.sdk.nexmark.Main
 task run(type: JavaExec) {
+  dependsOn ":beam-runners-google-cloud-dataflow-java-legacy-worker:shadowJar"
```
Comment (lukecwik) [inline sdks/java/testing/nexmark/build.gradle]: This should only be done if the Dataflow runner is being used. Note that nexmark is being used for many runners.

See `shouldProvideSpark`, you could copy the idea and add a `isDataflow` variable.

File: sdks/java/testing/nexmark/build.gradle [hunk 2]
```diff
@@ -21,11 +21,16 @@ applyJavaNature()
 
 description = "Apache Beam :: SDKs :: Java :: Nexmark"
 
+/*
+ * We need to rely on manually specifying these evaluationDependsOn to ensure that
+ * the following projects are evaluated before we evaluate this project. This is because
+ * we are attempting to reference the "sourceSets.test.output" directly.
+ */
+evaluationDependsOn(":beam-runners-google-cloud-dataflow-java-legacy-worker")
```
Comment (lukecwik) [inline sdks/java/testing/nexmark/build.gradle]: Only do this if this is for Dataflow

**Comment by HuangLED** [comment]
Thanks for reviewing (and merging)!

**Comment by lgajowy** [comment]
Are recent Dataflow + Nexmark  failures related to this change? See the logs: https://builds.apache.org/view/A-D/view/Beam/job/beam_PostCommit_Java_Nexmark_Dataflow/943/console 

And the error in particular: 

```
java.io.IOException: com.google.api.client.googleapis.json.GoogleJsonResponseException: 410 Gone
10:01:00 {
10:01:00   "code" : 429,
10:01:00   "errors" : [ {
10:01:00     "domain" : "usageLimits",
10:01:00     "message" : "The total number of changes to the object temp-storage-for-perf-tests/nexmark/staging/beam-runners-google-cloud-dataflow-java-legacy-worker-2.9.0-SNAPSHOT-VzyH4c3KuvZFQMINha03bw.jar exceeds the rate limit. Please reduce the rate of create, update, and delete requests.",
10:01:00     "reason" : "rateLimitExceeded"
10:01:00   } ],
10:01:00   "message" : "The total number of changes to the object temp-storage-for-perf-tests/nexmark/staging/beam-runners-google-cloud-dataflow-java-legacy-worker-2.9.0-SNAPSHOT-VzyH4c3KuvZFQMINha03bw.jar exceeds the rate limit. Please reduce the rate of create, update, and delete requests."
10:01:00 }
10:01:00 	at com.google.cloud.hadoop.util.AbstractGoogleAsyncWriteChannel.waitForCompletionAndThrowIfUploadFailed(AbstractGoogleAsyncWriteChannel.java:432)
10:01:00 	at com.google.cloud.hadoop.util.AbstractGoogleAsyncWriteChannel.close(AbstractGoogleAsyncWriteChannel.java:287)
10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.$closeResource(PackageUtil.java:260)
10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.tryStagePackage(PackageUtil.java:260)
10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.tryStagePackageWithRetry(PackageUtil.java:203)
10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.stagePackageSynchronously(PackageUtil.java:187)
10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.lambda$stagePackage$1(PackageUtil.java:171)
10:01:00 	at org.apache.beam.sdk.util.MoreFutures.lambda$supplyAsync$0(MoreFutures.java:104)
10:01:00 	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1626)
10:01:00 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
10:01:00 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
10:01:00 	at java.lang.Thread.run(Thread.java:748)
10:01:00 Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 410 Gone
```

**Comment by HuangLED** [comment]
> Are recent Dataflow + Nexmark failures related to this change? See the logs: https://builds.apache.org/view/A-D/view/Beam/job/beam_PostCommit_Java_Nexmark_Dataflow/943/console
> 
> And the error in particular:
> 
> ```
> java.io.IOException: com.google.api.client.googleapis.json.GoogleJsonResponseException: 410 Gone
> 10:01:00 {
> 10:01:00   "code" : 429,
> 10:01:00   "errors" : [ {
> 10:01:00     "domain" : "usageLimits",
> 10:01:00     "message" : "The total number of changes to the object temp-storage-for-perf-tests/nexmark/staging/beam-runners-google-cloud-dataflow-java-legacy-worker-2.9.0-SNAPSHOT-VzyH4c3KuvZFQMINha03bw.jar exceeds the rate limit. Please reduce the rate of create, update, and delete requests.",
> 10:01:00     "reason" : "rateLimitExceeded"
> 10:01:00   } ],
> 10:01:00   "message" : "The total number of changes to the object temp-storage-for-perf-tests/nexmark/staging/beam-runners-google-cloud-dataflow-java-legacy-worker-2.9.0-SNAPSHOT-VzyH4c3KuvZFQMINha03bw.jar exceeds the rate limit. Please reduce the rate of create, update, and delete requests."
> 10:01:00 }
> 10:01:00 	at com.google.cloud.hadoop.util.AbstractGoogleAsyncWriteChannel.waitForCompletionAndThrowIfUploadFailed(AbstractGoogleAsyncWriteChannel.java:432)
> 10:01:00 	at com.google.cloud.hadoop.util.AbstractGoogleAsyncWriteChannel.close(AbstractGoogleAsyncWriteChannel.java:287)
> 10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.$closeResource(PackageUtil.java:260)
> 10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.tryStagePackage(PackageUtil.java:260)
> 10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.tryStagePackageWithRetry(PackageUtil.java:203)
> 10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.stagePackageSynchronously(PackageUtil.java:187)
> 10:01:00 	at org.apache.beam.runners.dataflow.util.PackageUtil.lambda$stagePackage$1(PackageUtil.java:171)
> 10:01:00 	at org.apache.beam.sdk.util.MoreFutures.lambda$supplyAsync$0(MoreFutures.java:104)
> 10:01:00 	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1626)
> 10:01:00 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 10:01:00 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 10:01:00 	at java.lang.Thread.run(Thread.java:748)
> 10:01:00 Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 410 Gone
> ```

Hmmm...  But since the error is about worker jar, could be relevant. Looking. 

+ @boyuanzz as well

**Comment by HuangLED** [comment]
@lgajowy 

Looking at the error more closely.  The error happens within operation 

```com.google.cloud.hadoop.util.AbstractGoogleAsyncWriteChannel$UploadOperation```

rather than from inside worker jar. 

Maybe we are hitting testing account cloud usage , or started too many performance tests in parallel?

**Pull request merged**